### DISCUSSION TITLE
Using Path navigation when the feature view is already  contained in NavigationStack

### DISCUSSION THREAD

**Discussion by shannoga** [pointfreeco/swift-composable-architecture#3214] [Q&A] [answered]
Hi

We have a legacy App that we migrate to TCA.
As part of the App We have a menu that navigates to different features of our app. this menu is using a `NavigationStack` to push every feature as a destination. In some of our features we have an inner logics that needs it own `NavigationStack` and uses the Path pattern. 
`NavigationStack(path: $store.scope(state: \.path, action: \.path))`

That creates two `NavigationStack`s the later nested in the first one.

I am looking for away to create the destinations without creating a new `NavigationStack`, Looking inside I found this Modifier -

```
       .modifier(
            _NavigationDestinationViewModifier(store: path.wrappedValue, destination: destination)
          )
 ```
 
 I have tried to use it but it seems there is private stuff you are using there.
 I there a way to created the destinations for the Paths without using `NavigationStack(path:destinations:)`?
 
 Thanks a lot

**Comment by mbrandonw** [comment]
Hi @shannoga, I'm sorry but I do not really understand the question. A small sample project that demonstrates the problem would help.

**Reply by shannoga** [reply]
Thanks @mbrandonw 

This is the main view of the feature
```swift
     public var body: some View {
        
        WithPerceptionTracking {
           
            NavigationStack(path: $store.scope(state: \.path, action: \.path)) {
                ZStack {
                    switch store.initialState  {
                    case .giftsBuilder(let state):
                        SFGBuilderGiftsView(store:Store(
                            initialState: state,
                            reducer: {
                                SFGBuilderGiftsFeature()
                            }))
                    case .bookBuilder(let state):
                        SFGBuilderBookView(store:Store(
                            initialState: state,
                            reducer: {
                                SFGBuilderBookFeature()
                            }))
                    case .none:
                        ProgressView()
                    }
                }

            } destination: { store in
                switch store.case {
                case let .giftsBuilder(store):
                    SFGBuilderGiftsView(store: store)
                case let .bookBuilder(store):
                    SFGBuilderBookView(store: store)
                }
            }
        }
        .ignoresSafeArea(.keyboard, edges: .bottom)
    }
```

It is pushed from an existing NavigationViewController in the legacy app using a HostingViewController-
For Example -
```swift
navigationController.push(flowHostingViewController)
```

So we basically have a navigationViewController on the App itself and another NavigationViewController (NavigationStack) in the feature.
We want to know is there a way to create the destinations with out a navigationStack.

Hope it is clear now
Thanks!!

**Reply by mbrandonw** [reply, marked as answer]
Since navigation controllers/stacks cannot be nested, you simply cannot use `StackState` for your domain. You can use tree-based navigation, where views are pushed onto the stack via optional state, but that is quite different from how you have things structured now.

Or you could look at some of the tools we have built for our [UIKit navigation beta](https://github.com/pointfreeco/swift-composable-architecture/pull/3180) which allows you to use a stack-based controller in UIKit with TCA features and vanilla SwiftUI/UIKit features.

Unfortunately there is no easy answer here. You are going to have to do some work to disentangle things and have just a single navigation controller/view powering your features.

**Reply by shannoga** [reply]
I assume things won't be different if the legacy app was SwiftUI.
We will try to find a solution.
Thanks
